bentoml/utils/s3.py

```diff
@@ -33,54 +38,78 @@ def is_s3_url(url):
         return False
 
 
-def upload_to_s3(s3_url, file_path):
-    """
-    Update files in the file_path to the s3 location
-    """
+def upload_directory_to_s3(
+    upload_directory_path, region, bucket_name, s3_path_prefix=''
+):
+    s3_client = boto3.client('s3', region)
+    try:
+        for root, _, file_names in os.walk(upload_directory_path):
+            relative_path_to_upload_dir = os.path.relpath(root, upload_directory_path)
+            if relative_path_to_upload_dir == '.':
+                relative_path_to_upload_dir = ''
+            for file_name in file_names:
+                key = os.path.join(
+                    s3_path_prefix, relative_path_to_upload_dir, file_name
+                )
+                logger.debug(
+                    'Uploading {name} to s3 {location}'.format(
+                        name=file_name, location=bucket_name + '/' + key
+                    )
+                )
+                s3_client.upload_file(os.path.join(root, file_name), bucket_name, key)
+    except Exception as error:
+        raise BentoMLException(str(error))
```

Comment:
Add an error message prefix:

raise BentoMLException("Failed to upload directory to s3 bucket %s, error: %s", bucket_name, str(error))